### PR TITLE
book list and detail from API

### PR DESCRIPTION
**Pull request by damfinkel**
## Summary

Book list and detail from API.

Created a global AxiosInstance instead of creating one for each request.

## Screencast

df-training

### CODE REVIEW

File: src/components/BookDetail/BookDetail.js [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
 import BookSuggestions from './BookSuggestions';
 import BookComments from './BookComments';
 
 class BookDetail extends React.Component {
-  state = {
-    book: books.find(book => book.id === parseInt(this.props.match.params.id, 10)) // eslint-disable-line react/prop-types
+  state = {};
+
+  componentDidMount = () => {
+    AxiosInstance.get(`/books/${this.props.match.params.id}`) // eslint-disable-line react/prop-types
```
Comment (sbalay) [inline src/components/BookDetail/BookDetail.js]: Move the interfaces against external services to a service file. i.e: BookService

File: src/components/BookDetail/BookDetail.js [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+      .then(response => {
+        this.setState({ book: response.data });
+      })
+      .catch(() => {
+        localStorage.removeItem('token');
```
Comment (sbalay) [inline src/components/BookDetail/BookDetail.js]: you can use axios response interceptors to handle this cases in all the requests: https://github.com/axios/axios#interceptors

File: src/components/Dashboard/Dashboard.js [hunk 3]
```diff
@@ -10,9 +10,22 @@ class Dashboard extends React.Component {
...
     toBookDetail: false,
+    noAuth: false,
     currentFilterType: '',
     currentFilterValue: '',
-    filteredBooks: books
+    filteredBooks: []
+  };
+
+  componentDidMount = () => {
+    AxiosInstance.get('/books')
```
Comment (sbalay) [inline src/components/Dashboard/Dashboard.js]: service

File: src/components/Dashboard/Dashboard.js [hunk 4]
```diff
@@ -25,10 +38,10 @@ class Dashboard extends React.Component {
 
   filterBooks = () => {
     if (!this.state.currentFilterValue || !this.state.currentFilterType) {
-      this.setState({ filteredBooks: books });
+      this.setState({ filteredBooks: this.allBooks });
     } else {
       this.setState({
-        filteredBooks: books.filter(
+        filteredBooks: this.allBooks.filter(
```
Comment (sbalay) [inline src/components/Dashboard/Dashboard.js:43]: use the filters from the api. the query params for these are: `author` and `title`

File: src/components/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -10,9 +10,22 @@ class Dashboard extends React.Component {
...
+    filteredBooks: []
+  };
+
+  componentDidMount = () => {
+    AxiosInstance.get('/books')
+      .then(response => {
+        this.allBooks = response.data;
+        this.setState({ filteredBooks: response.data });
+      })
+      .catch(() => {
```
Comment (sbalay) [inline src/components/Dashboard/Dashboard.js:25]: use axios interceptors

**Review by sbalay — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/Dashboard.js (hunk 4 above)
Comment (damfinkel) [reply src/components/Dashboard/Dashboard.js:43]: I'll do it in the redux I'll do it in the redux for books branch I'm about to pull request if you don't mind

File: src/config/AxiosInstance.js [hunk 6]
```diff
@@ -4,6 +4,14 @@ const AxiosInstance = axios.create({
   baseURL: 'https://wbooks-api-stage.herokuapp.com/api/v1/'
 });
 
+AxiosInstance.interceptors.response.use(
+  config => config,
+  error => {
+    localStorage.removeItem('token');
+    Promise.reject(error);
```
Comment (sbalay) [inline src/config/AxiosInstance.js:11]: you should return this value

**Review by sbalay — APPROVED** [review]

**Pull request closed without merging**